src/decoders/image.rs

```diff
@@ -73,6 +78,19 @@ impl RawImage {
       camera.blacklevels
     };
 
+    // tuple format is top, right, bottom left
+    let mut blackarea: Vec<(u64,u64,u64,u64)> = Vec::new();
+
+    if camera.blackareah.1 != 0 {
```

Comment:
I had originally misunderstood blackareah and blackareav. I had thought that they formed the coordinates for a maked rectangle. I now see that blackareah is the vertical bound for a masked area that spans the entire width of the sensor, and blackareav is the horizontal bound for a masked area spanning the entire height of the sensor.

This means that the `blackarea` field will usually have two elements in the vector; one for the full width black area and one for the full height one. Incidentally, this matches the `MaskedAreas` description in the DNG spec.

Reply:
Yeah, a vector of 4 integers is the general way to just specify whatever regions you want. I simplified that because I had to write the masked area metadata by hand and simplified that internal API for convenience given that that was enough for the Canon cameras.